core/src/operations/mod.rs

```diff
@@ -441,6 +443,148 @@ pub enum Operation {
     RCombBase,
 }
 
+pub enum OperationData {
+    Felt(Felt),
+    U32(u32),
+    None,
+}
+
+/// Constructors
+impl Operation {
+    /// Builds an operation from its opcode and inline data (if any).
+    pub fn with_opcode_and_data(
+        opcode: u8,
+        data: OperationData,
+    ) -> Result<Self, DeserializationError> {
+        match opcode {
+            0b0000_0000 => Ok(Self::Noop),
+            0b0000_0001 => Ok(Self::Eqz),
+            0b0000_0010 => Ok(Self::Neg),
```

Comment:
This is rather error-prone. I wonder if we should just assign discriminant values in the `Operation` enum itself. This would allow us to get rid of this method and also `Operation::op_code()` would become obsolete.

If the above changes make sense, I probably would do them in a separate PR, merge that PR, and then rebase this PR from that one.

Reply:
#1390